src/firmware/firmware.ino

```diff
@@ -2127,8 +2141,11 @@ void handleClick(int event, SoundType soundType) {
     case 6:
       nightMode = !nightMode;
       if (nightMode) {
+
         prevBrightness = settings.brightness;
       }
```

Comment:
This code should be moved to `saveHaNightMode` function to have identical behavior for button and HA night mode activation
```
nightMode = !nightMode;
if (nightMode) {
    prevBrightness = settings.brightness;
}
```